src/Middleware/DetectLocaleMiddleware.php

```diff
@@ -172,15 +167,101 @@ public function setPersistLocale(HTTPRequest $request, $locale, $key = null)
     /**
      * Get the Fluent locale persistence key. See the "persist_ids" config static.
      *
-     * @param  string|null $key
      * @return string
      */
-    public function getPersistKey($key = null)
+    public function getPersistKey()
```

Comment:
Seeing as the methods below are now protected, do you think we should make these other methods protected as well and just leave `process` as public? We can probably just have integration tests for this class instead of smaller units